test/framework/shootmigrationtest.go

```diff
@@ -119,6 +119,10 @@ func (t *ShootMigrationTest) MigrateShoot(ctx context.Context) error {
 
 // GetNodeNames uses the shootClient to fetch all Node names from the Shoot
 func (t *ShootMigrationTest) GetNodeNames(ctx context.Context, shootClient kubernetes.Interface) (nodeNames []string, err error) {
+	if t.Shoot.Status.IsHibernated {
```

Comment:
Don't we need to consider both spec and status? There might be the case when hibernation is triggered just now and then we will have `spec.hibernation.enabled=true` and `.status.isHibernated=false`.

```suggestion
	if gardencorev1beta1helper.HibernationIsEnabled(t.Shoot) || t.Shoot.Status.IsHibernated {
```

Reply:
Hmm, this case was actually never considered, as a possible scenario, or at least not for this PR. @plkokanov  do you think that we need to address this possibility?

Reply:
@ialidzhikov I would leave this for a future PR as I'm currently running this scenario manually and checking any corner cases that arise. Currently we haven't thought for any additional changes to the integration test that might be required.

Reply:
Okay, no objections from my side, we can consider it in the future.
I just wanted to point out that most of the places in Gardener and in the extension library check both `spec.hibernation.enabled` and `status.isHibernated` when considering whether Shoot is hibernated or not.